core/src/storage2/collections/binary_heap/reverse.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! [`BinaryHeap`](`super::BinaryHeap`) is a max-heap by default, where the *largest* element will
+//! be returned by `heap.pop()`. To use a [`BinaryHeap`](`super::BinaryHeap`) as a min-heap, where
+//! the *smallest* element returned by `heap.pop()`, the type `T` of the binary tree can be wrapped
+//! in a `Reverse<T>`.
+//!
+//! [`Reverse`] simply wraps [`core::cmp::Reverse`] and implements all the required traits for use
+//! as a storage struct.
+
+use crate::storage2::traits::{
+    KeyPtr,
+    PackedLayout,
+    SpreadLayout,
+};
+use ink_prelude::vec::Vec;
+use ink_primitives::Key;
+
+/// Wrapper for [`core::cmp::Reverse`] for using a [`BinaryHeap`](`super::BinaryHeap`) as a min-heap
```

Comment:
Nitpick, decide for yourself, but for consistency reasons I would appreciate having one line comments end with `.` (there are other instances in this file as well).